tests/functional/chalk/runner.py

```diff
@@ -358,6 +358,7 @@ def run(
         tty: bool = False,
         show_config: bool = False,
         inject_binary_into_zip: bool = False,
+        zip_allowed_extensions: Optional[list[str]] = None,
```

Comment:
dont think this needs to be a top-level param here.

for more obscure configs if necessary we can pass a custom config file for tests

there is a whole folder for those already:

https://github.com/crashappsec/chalk/tree/main/tests/functional/data/configs

Reply:
reduced this down two only two additional test cases